boards/st/stm32h757i_eval/doc/index.rst

```diff
@@ -0,0 +1,298 @@
+- 1-Gbit twin Quad-SPI NOR flash memory or two 512-Mbit Quad-SPI NOR flash memories
+- Potentiometer
+- 4 color user LEDs
+- Reset, wakeup, tamper, or key buttons
+- Joystick with 4-direction control and selector
+
+
+More information about the board can be found at the `STM32H757I-EVAL website`_.
+More information about STM32H757XIH6 can be found here:
+
+- `STM32H757XI on www.st.com`_
+- `STM32H757xx reference manual`_
+- `STM32H757xx datasheet`_
+
+Supported Features
+==================
+
+The Zephyr  ``stm32h757i_eval`` board supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| ETHERNET  | on-chip    | ethernet                            |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+| FMC       | on-chip    | memc (SDRAM)                        |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| QSPI NOR  | on-chip    | off-chip flash                      |
++-----------+------------+-------------------------------------+
+| SDMMC     | on-chip    | disk access                         |
++-----------+------------+-------------------------------------+
+| IPM       | on-chip    | virtual mailbox based on HSEM       |
++-----------+------------+-------------------------------------+
+| DISPLAY   | on-chip    | MIPI DSI Host with shield (MP1166)  |
+|           |            | st_b_lcd40_dsi1_mb1166              |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+The default configuration per core can be found in the defconfig files:
+:zephyr_file:`boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m7_defconfig` and
+:zephyr_file:`boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m4_defconfig`
```

Comment:
Please replace with the following and let magic fill out the table for you :) (see e.g. https://docs.zephyrproject.org/latest/boards/st/stm32h747i_disco/doc/index.html#supported-features)

```suggestion
.. zephyr:board-supported-hw::
```